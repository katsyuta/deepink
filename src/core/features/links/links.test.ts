import {
	findLinksInText,
	formatNoteLink,
	formatResourceLink,
	getAppResourceDataInUrl,
} from '.';

test('formatters and parsers behavior are consistent', () => {
	const link = formatNoteLink('foo-bar');
	// eslint-disable-next-line spellcheck/spell-checker
	const link2 = formatResourceLink('baz-qux');

	const exampleText = `Example text with links. See a link ${link} on note and link ${link2} on resource`;

	const linksInText = findLinksInText(exampleText);
	expect(linksInText).toMatchSnapshot();

	const parsedLinks = linksInText.map((link) => getAppResourceDataInUrl(link.url));
	expect(parsedLinks).toEqual([
		{
			id: 'foo-bar',
			type: 'note',
		},
		{
			// eslint-disable-next-line spellcheck/spell-checker
			id: 'baz-qux',
			type: 'resource',
		},
	]);
});
