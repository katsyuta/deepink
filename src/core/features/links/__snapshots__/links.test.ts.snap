// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatters and parsers behavior are consistent 1`] = `
Array [
  Object {
    "index": 36,
    "url": "note://foo-bar",
  },
  Object {
    "index": 68,
    "url": "res://baz-qux",
  },
]
`;
